src/index/hook_maintain.rs

```diff
@@ -0,0 +1,223 @@
+use crate::error::*;
+use crate::index::utils::get_handle;
+use crate::ipc::client;
+use crate::utils::cells::PgRefCell;
+use base::search::Handle;
+use pgrx::pg_sys::Oid;
+use std::collections::BTreeMap;
+use std::ptr::NonNull;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum TransactionIndex {
```

Comment:
1. I think it does not work for subtransactions.
2. This never happens because transaction B cannot see `CREATE INDEX` in transaction A.
3. I do not understand what do you mean.